Apatira_Authman/Assignments/FormValidation/Form_Validation.py

```diff
@@ -0,0 +1,41 @@
+from flask import Flask, render_template, request, redirect, session, flash
+import datetime
+import re
+import random
+
+app = Flask(__name__)
+app.secret_key = 'rocksteady'
+
+Email_REGEX = re.compile(r'[^\s*$][1,120]')
+
+@app.route('/')
+def index():
+	return render_template('FormValidation.html')
+
+@app.route('/process', methods=['POST'])
+def process():
+	if not 'name' in session:
+		session['name'] = ''
+	if not 'comment' in session:
+		session['comment'] = ''
+	if not 'location' in session:
+		session['location'] = ''
+	if not 'fav' in session:
+		session['fav'] = ''
```

Comment:
There's no need to do this (the lines above). imagine I give you a program:

```
session['x'] = ''
session['x'] = 'test'

```
The first line is completely useless, since we set the session['x'] variable to something in the 2nd line, without ever reading it's first assigned value. you only ned to check to see if a key is in a dictionary before accessing (reading) from it, not when you're setting it.